vue-custom-bootstrap/src/components/TextInput.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <b-form-group
+    :description="hasError ? '' : description"
+    :invalid-feedback="errorText"
+    :state="hasError()"
+    :label-for="id"
+    class="mb-4"
+  >
+    <label>
+      <span :class="[ subLabel ? 'font-weight-bold': '', 'col-form-label']">{{label}}</span>
+      <div class="col-form-label col-form-label-sm">{{subLabel}}</div>
+      <b-form-input
+        :id="id"
```

Comment:
To make our custom TextInput component work like a plain 'input' html element, we have to make sure we bind "value" and emit an input event, just like a has.

 We also do v-bind="$attrs" which is some magic that lets us automatically apply all the props attributes that could used on the component. For example automatically knows that the placeholder should be applied to the b-form-input inside the component.